packages-backend/express/src/middlewares/session-middleware.spec.ts

```diff
@@ -51,8 +56,49 @@ describe.each`
           'x-mc-api-forward-to-version': 'v2',
         },
         originalUrl: '/foo/bar',
+        ...options?.requestOptions,
       };
       const fakeResponse = {};
+
+      return { sessionMiddleware, fakeRequest, fakeResponse };
+    }
+
+    it('should verify the token and attach the session info to the request', async () => {
+      const { sessionMiddleware, fakeRequest, fakeResponse } = setupTest();
+
+      await new Promise<void>((resolve, reject) => {
+        // @ts-ignore
+        sessionMiddleware(fakeRequest, fakeResponse, (error) => {
```

Comment:
Nit: maybe it would make it easier to read in tests if we'd have a `function promisifyMiddleware` which takes a middleware does that we do here. Then we could use it here to await a `promisifiedMiddleware` and below. Just thinking out loud while reading the code :)

Reply:
Thanks for the suggestions, @tdeekens.

I had thought about it but wasn't sure it was needed.
```
function runSessionMiddleware(middleware: unknown, request: unknown, response: unknown) {
  return new Promise<void>((resolve, reject) => {
    // @ts-ignore
    middleware(request, response, (error) => {
      if (error) reject(error);
      else resolve();
    });
  });
}

it('...', async () => {
  const { sessionMiddleware, fakeRequest, fakeResponse } = setupTest();

  await runSessionMiddleware(sessionMiddleware, fakeRequest, fakeResponse);

  expect(...);
});
```
Do you think something like that would be more readable?

Reply:
Subjectively to me when I _first_ read the code yes. Upon _second_ reading not hat much anymore. In the end it's just moving stuff around and the "complexity" is somewhere. Still a `await runSessionMiddleware` or `await waitForSessionMiddleware` would make the `it` test case itself more learn which is something I personally like.

Totally up to you :)

Reply:
I like the `waitForSessionMiddleware` naming so I went that path.

Updated in this commit f41c00b7da421e517ed093abfe4c1d357062e552

Reply:
🙇🏼 thank you!